mapusaurus/api/views.py

```diff
@@ -1,24 +1,84 @@
 import json
 
+from django.db.models import Count
 from django.http import HttpResponse, HttpResponseBadRequest
-from geo.views import tract_centroids_as_json 
-from censusdata.views import race_summary_as_json
-from hmda.views import loan_originations_as_json
+from geo.views import tract_centroids_as_json, get_censustract_geoids, get_censustract_geos
+from geo.models import Geo
+from censusdata.views import race_summary_as_json, minority_aggregation_as_json
+from hmda.models import HMDARecord
+from hmda.views import loan_originations_as_json, loan_originations
 from respondents.views import branch_locations_as_json
+from django.views.decorators.cache import cache_page
 
 def all(request):
     """This endpoint allows multiple statistical queries to be made in a
     single HTTP request"""
     try:
-
         hmda = loan_originations_as_json(request)
-
         minority = race_summary_as_json(request)
-        responses = {'minority' : minority, 'loanVolume': hmda }
+        responses = {'minority' : minority, 'loanVolume': hmda}
         return HttpResponse(json.dumps(responses), content_type='application/json')
     except:
         return HttpResponseBadRequest("invalid endpoint")
 
+def tables(request):
+    try:
+        table_data = minority_aggregation_as_json(request)
+        context = {'table_data': table_data}
+        return HttpResponse(json.dumps(context), content_type='application/json')
+    except:
+        return HttpResponseBadRequest("the following request failed: %s" % request)
+
+def msas(request):
+    try:
+        query = get_censustract_geos(request, metro=True)
+        msas = {
+        'msas': [msa.geoid for msa in query]
+        }
+        return HttpResponse(json.dumps(msas), content_type='application/json')
+    except:
+        return HttpResponseBadRequest("request failed; details: %s" % request)
+
+def msa(request):
+    try:
+        metro = request.GET.get('metro')
+        tracts = Geo.objects.filter(geo_type=Geo.TRACT_TYPE, cbsa=metro)
+        tract_loans = loan_originations_as_json(request)
+    except:
+        return HttpResponseBadRequest("request failed; details: %s" % request)
+    else:
+        try:
+            with open("/var/www/static/tracts/%s.json" % metro, 'r') as f:
```

Comment:
Instead of writing and loading from disk, consider using the django low level caching api. That'll allow you to swap out the backend for file system, memcached, etc. etc. by configuration.